File: backend/src/main/java/com/sga/service/DirectivoService.java
package com.sga.service;

import com.sga.dto.EvaluacionDTO;
import com.sga.dto.EstudianteDTO;
import com.sga.dto.GrupoDTO;
import com.sga.dto.HojaDeVidaDTO;

import java.util.List;
import java.util.UUID;

public interface DirectivoService {
    List<GrupoDTO> listarGruposPorGrado(UUID idGrado);
    List<EstudianteDTO> listarEstudiantesDeGrupo(UUID idGrupo);
    HojaDeVidaDTO obtenerHojaDeVida(UUID idEstudiante);
    HojaDeVidaDTO actualizarHojaDeVida(UUID idEstudiante, HojaDeVidaDTO hojaDeVidaDTO);
    List<EvaluacionDTO> obtenerHistoricoLogros(UUID idEstudiante);
}

File: frontend/lib/services/directivo.service.ts
import { api } from '../api'

export interface GradoDTO {
  idGrado: string
  nombreGrado: string
}

export interface GrupoDTO {
  idGrupo: string
  nombre: string
  idGrado: string
  nombreGrado: string
  idProfesor?: string
  nombreProfesor?: string
  cantidadEstudiantes: number
}

export interface EstudianteDTO {
  idEstudiante: string
  nombre: string
  nombre2?: string
  apellido: string
  apellido2?: string
  numeroDocumento: string
  estado?: boolean
  idGrado?: string
  nombreGrupo?: string
  gradoSolicitado?: string
}

export interface HojaDeVidaDTO {
  idHojaDeVida: string
  idEstudiante: string
  nombreCompleto: string
  nombreAcudiente?: string
  fechaNacimiento: string
  documentoIdentidad: string
  detallesMedicos: string[]
  observacionesAprendizaje: string[]
}

export const DirectivoService = {
  /**
   * Obtiene todos los grados disponibles
   */
  async obtenerGrados(): Promise<GradoDTO[]> {
    return await api.get<GradoDTO[]>('/grados')
  },

  /**
   * Obtiene los grupos de un grado específico
   */
  async obtenerGruposPorGrado(idGrado: string): Promise<GrupoDTO[]> {
    return await api.get<GrupoDTO[]>(`/directivo/grado/${idGrado}/grupos`)
  },

  /**
   * Obtiene los estudiantes de un grupo específico
   */
  async obtenerEstudiantesDeGrupo(idGrupo: string): Promise<EstudianteDTO[]> {
    return await api.get<EstudianteDTO[]>(`/directivo/grupo/${idGrupo}/estudiantes`)
  },

  /**
   * Obtiene la hoja de vida de un estudiante
   */
  async obtenerHojaDeVida(idEstudiante: string): Promise<HojaDeVidaDTO> {
    return await api.get<HojaDeVidaDTO>(`/directivo/estudiante/${idEstudiante}/hoja-vida`)
  },

  /**
   * Actualiza la hoja de vida de un estudiante
   */
  async actualizarHojaDeVida(idEstudiante: string, hojaDeVida: HojaDeVidaDTO): Promise<HojaDeVidaDTO> {
    return await api.put<HojaDeVidaDTO>(`/directivo/estudiante/${idEstudiante}/hoja-vida`, hojaDeVida)
  }
}
